docs/sphinx/source/reference/sql_types.rst

```diff
@@ -72,14 +72,172 @@ In this example, `c` is an array, and each record within the array is a struct o
+
+    -- Insert a FLOAT precision vector
+    INSERT INTO embeddings VALUES (2, !! !v32 [0.5, 1.2, -0.8] !!);
+
+    -- Insert a DOUBLE precision vector
+    INSERT INTO embeddings VALUES (3, !! !v64 [0.5, 1.2, -0.8] !!);
+
+Casting Arrays to Vectors
+--------------------------
+
+While vector literals are not supported, you can use :sql:`CAST` to convert array expressions to vectors. The source array elements can be of any numeric type (:sql:`INTEGER`, :sql:`BIGINT`, :sql:`FLOAT`, :sql:`DOUBLE`):
+
+.. code-block:: sql
+
+    -- Cast FLOAT array to FLOAT vector
+    SELECT CAST([1.2, 3.4, 5.6] AS VECTOR(3, FLOAT)) AS vec;
+
+    -- Cast INTEGER array to HALF vector
+    SELECT CAST([1, 2, 3] AS VECTOR(3, HALF)) AS vec;
+
+    -- Cast mixed numeric types to DOUBLE vector
+    SELECT CAST([1, 2.5, 3L] AS VECTOR(3, DOUBLE)) AS vec;
+
+The array must have exactly the same number of elements as the vector's declared dimension, or the cast will fail with error code :sql:`22F3H`. Only numeric arrays can be cast to vectors.
+
+Querying Vectors
+----------------
+
+Vectors can be selected and compared like other column types. When comparing vectors in WHERE clauses, you would typically use prepared statement parameters in your Java/JDBC code:
+
+.. code-block:: sql
+
+    -- Select vectors
+    SELECT embedding FROM embeddings WHERE id = 1;
+
+    -- Compare vectors for equality (using test DSL syntax; in actual code use PreparedStatement)
+    SELECT id FROM embeddings WHERE embedding = !! !v32 [0.5, 1.2, -0.8] !!;
+
+    -- Compare vectors for inequality
+    SELECT id FROM embeddings WHERE embedding != !! !v32 [1.0, 2.0, 3.0] !!;
+
+    -- Check for NULL vectors
+    SELECT id FROM embeddings WHERE embedding IS NULL;
+    SELECT id FROM embeddings WHERE embedding IS NOT NULL;
+
+    -- Use IS DISTINCT FROM for NULL-safe comparisons
+    SELECT embedding IS DISTINCT FROM !! !v32 [0.5, 1.2, -0.8] !! FROM embeddings;
+
+Vectors in Struct Fields
+-------------------------
+
+When vectors are nested within struct types, you can access them using dot notation. As with all vector comparisons, you would use prepared statement parameters in your JDBC code:
+
+.. code-block:: sql
+
+    -- Access vector within a struct
+    SELECT embedding.embedding FROM documents WHERE id = 1;
+
+    -- Filter by vector within struct (using test DSL syntax; in actual code use PreparedStatement)
+    SELECT id FROM documents
+    WHERE embedding.embedding = !! !v32 [0.5, 1.2, -0.8] !!;
+
+    -- Check NULL for vector field in struct
+    SELECT id FROM documents WHERE embedding.embedding IS NULL;
+
+Supported Operations
+====================
+
+The following operations are supported on vector types:
+
+* **Equality comparison** (:sql:`=`, :sql:`!=`)
+* **NULL checks** (:sql:`IS NULL`, :sql:`IS NOT NULL`)
+* **NULL-safe comparison** (:sql:`IS DISTINCT FROM`, :sql:`IS NOT DISTINCT FROM`)
+* **CAST from numeric arrays** to vectors
```

Comment:
Should there be a mention here of the fact that you cannot create value indexes on vectors? 
We also probably want to mention that you cannot ORDER BY a vector field at this point. 
Can we group by a vector field?